rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+instants i.e. over instant ranges where cleaner has not been executed. So, if we guarantee that 1.x active timeline will definitely contain all the uncleaned instants, we could get away by not even porting over 
+LSM timeline reader logic to 0.16.0. 
+
+#### Completion time based read in FileGroup/FileSlice grouping and log block reads
+- As of this write up, we are not supporting completion time based log file ordering in in 0.16. Punting it as called out earlier.
+- What's the impact if not for this support:
+    - For users having single writer mode or OCC in 1.x, 0.16.0 reader not supporting completion time based read(log file ordering) should still be ok. Both 1.x reader and 0.16.0 reader should have same behavior.
+    - But only if someone has NBCC writes and have log files in different ordering written compared to the log file versions, 0.16.0 reader might result in data consistency issues. but since we are calling out that 0.16.0 is a bridge release and recommend users to migrate all the readers to 1.x fully before starting to enable any new features for 1.x tables.
+    - Example scenarios. say, we have lf1_10_25, lf2_15_20(format "logfile[index]_[starttime]_[completiontime]") for a file slice. In 1.x reader, we will order and read it as lf2 followed by lf1. w/o this support in 0.16.0, we might read lf1 followed by lf2. Just to re-iterate this might only impact users who have enabled NBCC and having multi writers writing log files in different ordering. Even if they were using OCC, one of the writers is expected to have failed (on the writer side) since data is overlapping for two writers in 1.x writer.
+
+### FileSystemView:
+- Support ignoring partially failed log files from FSV. In 0.16.0, from FSV standpoint, all log files(including partially failed) are valid. We let the log record reader ignore the partially failed log files. But
+  in 1.x, log files could be rolledback (deleted) by a concurrent rollback. So, the FSV should ensure it ignores the uncommitted log files. 
+- We don't need the completion time logic ported over either for file slice determination nor for log file ordering.\
+    - So, here is how File Slice determination will happen using 0.16.0 reader.\
+      a. Read base files and assign to resp file groups. \
+      b. Read log files. Parse instant time from log file name (it could refer to base instant time for a file written in 0.16.x, or it could refer to delta commit in case of 1.x writer). Find largest (<=) base instant times in the corresponding file group
+      and assign the log file to it. \
+      d. Log files within a file slice are ordered based on log version and write tokens.\
+      The same logic will be used whether we are reading a 0.16.x table or its a 1.x table. Only difference wrt how a 1.x reader will behave in comparison to 0.16.x reader while reading a 1.x table is when NBCC is involved w/ multi-writers. But as of this writing, 
```

Comment:
In summary, there is no additional change for file slicing if we just port the 1.x file slicing into 0.16.x, it should work fine.

Reply:
relaying the same response I added above. 
```
we don't need completion time in 0.16.0 reader if we do not need to support NBCC. as per feedback, I am going to add a section in this RFC on bridge release design if we were to support NBCC as well. To get that working, we will be needing completion time query view.
```